anchor/common/qbft/src/lib.rs

```diff
@@ -696,9 +696,23 @@ where
             return;
         }
 
-        // Make sure that we have accepted a proposal for this round
+        // Handle commit message, checking proposal acceptance and catch-up scenarios.
+
+        // If we have NOT accepted a proposal for this round, this is a catch-up scenario.
+        // We allow commits without having seen a proposal in this case.
         if !self.proposal_accepted_for_current_round {
-            warn!(from=?operator_id, ?self.state, "Have not accepted Proposal for current round yet");
+            debug!(from=?operator_id, ?self.state, "Have not accepted Proposal for current round yet (catch-up scenario)");
+            return;
```

Comment:
Inconsistent states are treated all the same with just a return; is this good enough?